--- barcode-0.99/plessey.c.orig	2014-05-04 13:23:01.392919441 +0300
+++ barcode-0.99/plessey.c	2014-05-04 13:24:55.204785523 +0300
@@ -149,7 +149,7 @@
                         checkptr[i+j] ^= check[j];
     }
     for (i = 0; i < 8; i++) {
-        sprintf(ptr, patterns[checkptr[strlen(text) * 4 + i]]);
+        sprintf(ptr, "%s", patterns[checkptr[strlen(text) * 4 + i]]);
 	ptr += 2;
     }
     fprintf(stderr, _("CRC: "));
